test/specs/palindrom.errors.test.js

```diff
@@ -1,331 +1,291 @@
+                });
+
+                await sleep();
+
+                // make sure WS is up
+                assert.equal(palindrom.network._ws.readyState, 1);
 
-        const spy = sinon.spy();
+                assert(spy.calledOnce);
 
-        const palindrom = new Palindrom({
-          remoteUrl: 'http://localhost/testURL',
-          onOutgoingPatchValidationError: spy,
-          onStateReset(obj) {
-            obj.val = Number.MAX_SAFE_INTEGER + 1;
-          }
-        });
+                const errorPassed = spy.getCall(0).args[0];
 
-        setTimeout(() => {
-          assert(spy.calledOnce);
-          const errorPassed = spy.getCall(0).args[0];
-          assert(errorPassed instanceof RangeError);
-          assert.equal(
-            errorPassed.message,
-            `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER +
-              1}, variable path is: /val`
-          );
-          done();
-        }, 15);
-      });
-      it('Outgoing socket patches: out of range numbers should call onOutgoingPatchValidationError with a RangeError', function(
-        done
-      ) {
-        const server = new MockSocketServer('ws://localhost/testURL');
-
-        moxios.stubRequest('http://localhost/testURL', {
-          status: 200,
-          headers: { location: 'http://localhost/testURL' },
-          responseText: '{"val": 100}'
-        });
+                assert(errorPassed instanceof RangeError);
 
-        var spy = sinon.spy();
+                assert.equal(
+                    errorPassed.message,
+                    `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: 9007199254740992, variable path is: /val`
+                );
 
-        var palindrom = new Palindrom({
-          remoteUrl: 'http://localhost/testURL',
-          useWebSocket: true,
-          onOutgoingPatchValidationError: spy
-        });
+                server.stop();
+            });
+            it('Incoming socket patch: out of range numbers should call onIncomingPatchValidationError event with a RangeError', async () => {
+                const server = new MockSocketServer(getTestURL('testURL', false, true));
 
-        setTimeout(() => {
-          palindrom.obj.value = Number.MAX_SAFE_INTEGER + 1;
-          // make sure WS is up
-          assert.equal(palindrom.network._ws.readyState, 1);
-          assert(spy.calledOnce);
-          const errorPassed = spy.getCall(0).args[0];
-          assert(errorPassed instanceof RangeError);
-          assert.equal(
-            errorPassed.message,
-            `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER +
```

Comment:
Why is the value hardcoded here? I'd use interpolated one, as above

Reply:
I don't know much about JS engine internals, but doing math on integers above `MAX_SAFE_INTEGER`  sounds, well, *unsafe* to me. I would use the hardcoded value in the assertion just to be on the safe side.

Reply:
I think adding to `MAX_SAFE_INTEGER` is not a problem, the "unsafe" here means unable to compare.
And we do the math at `palindrom.obj.value = Number.MAX_SAFE_INTEGER + 1;` anyway.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
> Safe in this context refers to the ability to represent integers exactly and to correctly compare them. For example, Number.MAX_SAFE_INTEGER + 1 === Number.MAX_SAFE_INTEGER + 2 will evaluate to true, which is mathematically incorrect. See Number.isSafeInteger() for more information.